themes/default/content/blog/mlops-python-iac-deploy-a-chatbot-in-30-minutes/index.md

```diff
@@ -0,0 +1,197 @@
+# !WARNING! Please use a more secure password than this example.
+export PULUMI_CONFIG_PASSPHRASE_FILE=$HOME/.pulumi/secret
+echo "keepItSecretKeepItSafePassword" > ~/.pulumi/secret
+
+# There are many ways to store pulumi state, here we use a local file.
+# Other state backends include s3, Pulumi Cloud, and more.
+pulumi login file://~/.pulumi
+
+# Initialize your stack
+# Here we name the stack "dev"
+pulumi stack init --stack dev
+```
+
+
+### 3. Configure required credentials
+
+Configure your Huggingface credentials. If you open the `Pulumi.${stackName}.yaml` file, you can see the encrypted secrets in this plain text yaml configuration.
+
+```bash
+# Set Huggingface.co username
+pulumi config set hfUsername <huggingface_username>
+
+# Configure token as secret
+pulumi config set --secret hfToken <huggingface_api_token>
+
+# Set the model that you want to download
+pulumi config set hfModel "meta-llama/Llama-2-7b-chat-hf"
+```
+
+### 4. Deploy in the cloud on Runpod.io
+
+While there are 3 providers already supported in this codebase, I am going to demonstrate deploying to Runpod.io, a GPU and ML dedicated cloud catering to AI workloads, and development tasks.
+
+```bash
+# This enables the IaC to deploy Katwalk Server
+pulumi config set deploy True
+
+# Set the deploy runtime to runpod to deploy on the Runpod cloud
+pulumi config set runtime runpod
+
+# Configure your Runpod API Token as a secure Pulumi secret
+pulumi config set --secret runpodToken <your_runpod_api_token>
+
+# Finally, run `pulumi up` to deploy!
+pulumi up
+
+# When done, you can 'destroy' the stack to deprovision your deployment
+pulumi destroy
+```
+
+Providers supported so far include:
+
+* Docker (locally)
+* Runpod.io GPU Cloud
+* Azure Container Instances
+
+### 5. Build the Katwalk Container
+
+While there are many other ways to orchestrate ML services in the cloud and locally, in this project we are taking ownership of the full application and infrastructure story. In fact, all 3 supported providers use the same container image to serve the LlaMa LLM API Service, and you can build it yourself just like this:
+
+```bash
+# Set image build to True
+pulumi config set imageBuild True
+
+# Let's build this without pushing to a registry for now
+pulumi config set imagePush False
+
+# Give your katwalk image a project name
+pulumi config set registryProject $(whoami)
+
+# Now build the image!
+pulumi up
+```
+
+### 6. Talk to your chatbot!
+
+Once your Katwalk LlaMa server is running, you can interact with the API by sending prompts as a json payload. In future posts we will orchestrate a chat frontend for our chatbot but for now we'll use a curl command. You can also substitute an API developer tool such as Postman or Insomnia!
+
+```bash
+# send a prompt with curl
+curl -s -X 'POST' 'http://${RUNPOD_FQDN}:8000/v1/chat' -H 'accept: application/json' -H 'Content-Type: application/json' -d '{"prompt": "Write a short story about a robot."}'
```

Comment:
Not life or death, but is it worth considering formatting the curl request so that the users can see the prompt without having to scroll over?

Reply:
good feedback, yes